docs/_source/guides/llms/practical_guides/fine_tune_others.md

```diff
@@ -0,0 +1,194 @@
+    text=dataset.field_by_name("my_text_field"),
+    label=label_unification
+)
+```
+
+#### Use ArgillaTrainer
+
+Next, we can use our `FeedbackDataset` and `TrainingTaskMapingForTextClassification` to initialize our `argilla.ArgillaTrainer`. We support the frameworks "openai", "setfit", "peft", "spacy" and "transformers".
+
+````{note}
+This is a newer version and can be imported via `from argilla import ArgillaTrainer`. The old trainer can be imported via `from argilla.training import ArgillaTrainer`. Our docs, contain some [additional information on usage of the ArgillaTrainer](/guides/train_a_model).
+````
+
+```python
+from argilla import ArgillaTrainer
+
+dataset = ...
+training_task_mapping = ...
+trainer = ArgillaTrainer(
+    dataset=dataset,
+    training_task_mapping=training_task_mapping,
+    framework="setfit",
+    fetch_records=False
+)
+trainer.update_config(num_train_epochs=2)
+trainer.train(output_dir="my_awesone_model")
+```
+
+````{note}
+The `FeedbackDataset` also allows for custom workflows via the `prepare_for_training()`-method.
+```python
+training_task_mapping = ...
+dataset = ...
+dataset.prepare_for_training(
+    framework="setfit",
+    training_task_mapping=training_task_mapping
+)
+```
+````
+
+### An end-to-end example
+
+Underneath, you can also find an end-to-end example of how to use the `ArgillaTrainer`.
+
+```python
+import argilla as rg
+
+dataset = rg.FeedbackDataset(
+    guidelines="Add some guidelines for the annotation team here.",
+    fields=[
+        rg.TextField(name="text", title="Human prompt"),
+    ],
+    questions =[
+        rg.LabelQuestion(
+            name="relevant",
+            title="Is the response relevant for the given prompt?",
+            labels=["yes","no"],
+            required=True,
+        )
+    ]
+)
+dataset.add_records(
+    records=[
+        rg.FeedbackRecord(
+            fields={"text": "What is your favorite color?"},
+            responses=[{"values": {"relevant": {"value": "no"}}}]
+        ),
+        rg.FeedbackRecord(
+            fields={"text": "What do you think about the new iPhone?"},
+            responses=[{"values": {"relevant": {"value": "yes"}}}]
+        ),
+        rg.FeedbackRecord(
+            fields={"text": "What is your feeling about the technology?"},
+            responses=[{"values": {"relevant": {"value": "yes"}}},
+                       {"values": {"relevant": {"value": "no"}}},
+                       {"values": {"relevant": {"value": "yes"}}}]
+        ),
+        rg.FeedbackRecord(
+            fields={"text": "When do you expect to buy a new phone?"},
+            responses=[{"values": {"relevant": {"value": "no"}}},
+                       {"values": {"relevant": {"value": "yes"}}}]
+        )
+
+    ]
+)
+
+label_unification = rg.LabelQuestionUnification(
+    question=dataset.question_by_name("relevant"),
+    strategy="majority"
+)
+
+training_task_mapping = rg.TrainingTaskMapingForTextClassification(
```

Comment:
Also the naming for `TrainingTaskMappingForTextClassification` seems too long to me, maybe we can shorten it as `TrainingMapping(task="text-classification", ...)` or something similar? WDYT?

Reply:
Yes I agree. I was a bit unsure about the naming. Similar to Unfications, Strategies etc.

Reply:
I think I prefer to have the mapping more explicit because each task might require a different field/question config.

Reply:
Hmm and what about providing those via `__init__` in a generic way e.g. `TrainMapping(fields=..., *args, **kwargs)`, and have some classmethods to automatically build the class with a certain configuration? e.g. `TrainMapping.for_text_classification()`

Reply:
Sure, sounds good. You mean `TrainMapping(task=..., *args, **kwargs)` right?